acquire/tools/decrypter.py

```diff
@@ -444,6 +461,22 @@ def main():
                     executor.shutdown(wait=True, cancel_futures=True)
                     break
 
+            if len(files) > 1:
+                results_string = textwrap.indent(
+                    "\n".join(
+                        f"{file} - {'Success' if success else 'Failed'}: {message}"
+                        for file, (success, message) in sorted(results.items())
+                    ),
+                    " • ",
+                )
+                (progress.console.log if progress else log.info)(
+                    f"\nDecrypt results (file - result: message):\n{results_string}"
+                )
+
+            if not all(results.values()) or tasks:
```

Comment:
If there are tasks left in the queue, do they have a result output? Should this case not be handled separately?

Reply:
This is needed for the exit code. Because I could think of a case where all the files that were processed were successful, but there are still tasks that needed to be processed. In that case, the run was still partly successful and therefore we need to return a 2 exit code (2, because any() will then always evaluate to True)

Reply:
I see that it would return:

- 1 if all processes fail, irrespective if there are tasks left
- 1 if there are no results, irrespective if there are tasks left
- 2 if at least 1 process succeeds and the others fail, irrespective if there are tasks left
- 2 if all processes succeed but there are tasks left

So complete fail == 1, partial success == 2.

It is difficult to see from the code that this is the result. So it should get a comment, but better rewrite it to be a bit easier to parse.

Reply:
I've rewritten it and added some comments